### ISSUE TITLE
[Number field] Should have role="spinbutton"

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2922]
### Provide a general summary of the issue here

Number field is currently a text field, which only allows numbers (at least when validating) and users can change values using arrow up or arrow down and has somewhat discrete ranges (e.g., number cant be negative or must be between N and M). The actions are not clear for users from the role. The number field is similar to <input type="number">, which allows numbers only, and users can use up or down arrow keys to change the value. Number field should have a role spinbutton, which would indicate the actions directly from the role. 

Personally I would also only allow users to type numbers.

### 🤔 Expected Behavior?

Users are aware of the input methods.

### 😯 Current Behavior

Users can discover the input methods through trial and error, but are not native to a text field.

### 💁 Possible Solution

Use role="spibutton", which indicates that the field allows for numbers and has increment/decrement using up/down arrow keys.

### 🖥️ Steps to Reproduce

1. Focus the element with a screen reader -> No indication of actions are available through role or otherwise (apart from some variant where a range is specified using aria-describedby).

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning
